src/test/java/bio/terra/workspace/integration/common/response/ErrorOrObjectResponse.java

```diff
@@ -0,0 +1,31 @@
+package bio.terra.workspace.integration.common.response;
+
+import io.vavr.control.Either;
+import org.springframework.http.HttpStatus;
+
+public class ErrorOrObjectResponse<S, T> {
+
+  private HttpStatus statusCode;
+  private Either<S, T> data;
+
+  public HttpStatus getStatusCode() {
+    return statusCode;
+  }
+
+  public void setStatusCode(HttpStatus statusCode) {
+    this.statusCode = statusCode;
+  }
+
+  public Either<S, T> getData() {
+    return data;
+  }
+
+  public void setErrorObject(S left) {
+    data = Either.left(left);
+  }
+
+  public void setResponseObject(T right) {
+    data = Either.right(right);
+  }
+
+}
```

Comment:
I like this a lot better with `Either`! (my Scala bias is showing)

I might add:
```
  public T getResponseObject() throws NoSuchElementException { return data.get(); }

  public S getErrorObject() throws NoSuchElementException { return data.getLeft(); }

  public boolean isResponseObject() { return data.isRight(); }

  public boolean isErrorObject() { return data.isLeft(); }
```

and then, callers such as `WorkspaceIntegrationTest.java` could do something like:
```
    Assertions.assertTrue(workspaceResponse.isResponseObject());
    CreatedWorkspace createdWorkspace = workspaceResponse.getResponseObject();
    Assertions.assertEquals(workspaceId.toString(), createdWorkspace.getId());
```